src/Progress/LinearProgress.js

```diff
@@ -0,0 +1,204 @@
+// @flow weak
+
+import React, { Component, PropTypes } from 'react';
+import { createStyleSheet } from 'jss-theme-reactor';
+import classNames from 'classnames';
+
+export const styleSheet = createStyleSheet('LinearProgress', (theme) => {
+  const { palette } = theme;
+  return {
+    root: {
+      display: 'block',
+      position: 'relative',
+      width: '100%',
+      overflow: 'hidden',
+      height: 5,
+      transform: 'translate(0, 0) scale(1, 1)',
+      backgroundColor: palette.primary[100],
+    },
+    rootBuffer: {
+      backgroundColor: 'transparent',
+    },
+    bar: {
+      position: 'absolute',
+      left: 0,
+      bottom: 0,
+      top: 0,
+      height: 5,
+      transition: 'transform 0.2s linear',
+      backgroundColor: palette.primary[500],
+    },
+    dashed: {
+      '&:before': {
+        content: '""',
+        position: 'absolute',
+        marginTop: 0,
+        height: 5,
+        width: '100%',
+        background:
+          `radial-gradient(${palette.primary[100]} 0%, ${palette.primary[100]} 16%, transparent 42%)`,
+        backgroundSize: '10px 10px',
+        backgroundPosition: '0px -23px',
+        animation: 'buffer 3s infinite linear',
+      },
+    },
+    indeterminateBar1: {
+      willChange: 'left, right',
+      animation: 'indeterminate-1 2.1s cubic-bezier(0.65, 0.815, 0.735, 0.395) infinite',
+    },
+    indeterminateBar2: {
+      willChange: 'left, right',
+      animation: 'indeterminate-2 2.1s cubic-bezier(0.165, 0.84, 0.44, 1) infinite',
+      animationDelay: '1.15s',
+    },
+    determinateBar1: {
+      transition: 'width .3s linear',
+    },
+    determinateBar2: {
+      display: 'none',
+    },
+    bufferBar1: {
+      transition: 'width .3s linear',
+      backgroundColor: palette.primary[100],
+    },
+    bufferBar2: {
+      transition: 'width .3s linear',
+    },
+    queryBar1: {
```

Comment:
Looking at the specification, the query animation seems to be the same one as the `indeterminate` mode in the opposite direction. It's not quite the case right now.